### ISSUE TITLE
public int Number is a public field.

### ISSUE THREAD

**Issue by rbwhitaker** [rbwhitaker/CSharpPlayersGuideEarlyAccess#419]
I have a faint memory of this being intentional. I'm not sure it works the same (or as simply) with a property or with a private field (that part I'm sure of). This was reported and I think even if I don't turn it into a property, at least deserves a response on Discord.

**Comment by rbwhitaker** [comment]
I can get this to happen with a property or with a field. It doesn't matter what you use. The property version actually happens a lot more often (something like 25% of the time instead of 10% of the time). Though I did end up changing it to run 100 threads in parallel, instead of just the main thread and another thread, to increase the frequency.

The original complaint (if you want to call it that) with this item was that this is a public field. The book does discourage using public fields pretty heavily. However, I didn't want people to think that it was unique to properties. It can happen to any variable, anywhere.

I think maybe the solution is to just acknowledge that it is a bit out of place, and otherwise leave it alone.

**Comment by rbwhitaker** [comment]
After further review, I decided that it is probably less than ideal that I make such a strong point about keeping fields private, but then violate that rule the second it is convenient to do otherwise. So I've made a change to the book to make the field be called `_number` and added a get-only property called `Number` that returns it. Several things over the "Thread Safety" section needed to be updated to facilitate that, but I think it is better for it now.
